### PR TITLE
write jobflow tutorial

### PR DESCRIPTION
**Pull request by QuantumChemist**
I started with writing the `jobflow` tutorial

To-Dos:

- [x] revise text
- [x] put hyperlinks for all external packages that are mentioned like `atomate2`, `custodian` etc.

### CODE REVIEW

**Comment by JaGeo** [comment]
@QuantumChemist Thank you so much!!

File: jobflow.ipynb [hunk 1]
```diff
@@ -1 +1,799 @@
...
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 15,
+   "outputs": [],
+   "source": [
+    "graph = to_mermaid(qe_flow, show_flow_boxes=True)\n",
```
Comment (QuantumChemist) [inline jobflow.ipynb]: Could someone who has QE set up, run these lines please?

**Comment by QuantumChemist** [comment]
I have finished the tutorial @JaGeo 😀

File: jobflow.ipynb (hunk 1 above)
Comment (JaGeo) [reply jobflow.ipynb]: You can simply install it via conda. I have done this as well

**Comment by JaGeo** [comment]
@QuantumChemist I will check it out next week and will contribute as well!

File: jobflow.ipynb (hunk 1 above)
Comment (QuantumChemist) [reply jobflow.ipynb]: Oh cool, didn't know this 😁

**Comment by QuantumChemist** [comment]
I can't get QE to take the right pseudopot path. I will try again on another day ^^

**Comment by jan-janssen** [comment]
Somehow my notifications were not up to date and I missed this. To find the pseudo potentials in the Github Continuous integration environment we set the following environment variable 
```
export ESPRESSO_PSEUDO=$(pwd)/espresso/pseudo
```
Unfortunately the Github Action is not executed in the home directory, so the `ESPRESSO_PSEUDO` environment variable is used to specify the right path. Does `jobflow` use the environment variables from the external environment? Or is it necessary to set the environment variables explicitly?

File: jobflow.ipynb (hunk 1 above)
Comment (jan-janssen) [reply jobflow.ipynb]: We have a wide range of opensource codes available on conda-forge. You can find the full list here - at least the packages I maintain https://github.com/jan-janssen/conda-forge-contribution . If you are interested to add any other code I am happy to help.

**Comment by QuantumChemist** [comment]
> Somehow my notifications were not up to date and I missed this. To find the pseudo potentials in the Github Continuous integration environment we set the following environment variable
> 
> ```
> export ESPRESSO_PSEUDO=$(pwd)/espresso/pseudo
> ```
> 
> Unfortunately the Github Action is not executed in the home directory, so the `ESPRESSO_PSEUDO` environment variable is used to specify the right path. Does `jobflow` use the environment variables from the external environment? Or is it necessary to set the environment variables explicitly?

Oohh I might try to specify the QE pp env variable like this next time! And I have to check how jobflow handles env variables in general.

File: adis_tools/schemas/qes_230310.xsd [hunk 2]
```diff
@@ -22,7 +22,7 @@ datecode 220603
       <element type="qes:outputType" name="output" minOccurs="0" />
       <element type="qes:cpstatusType" name="STATUS" minOccurs="0"/>
       <element type="qes:cptimestepsType" name="TIMESTEPS" minOccurs="0"/>
-      <element type="qes:statusType" name="exit_status"  minOccurs="0" maxOccurs="1"/>
+      <element type="qes:statusType" name="status"  minOccurs="0" maxOccurs="1"/>
```
Comment (QuantumChemist) [inline adis_tools/schemas/qes_230310.xsd]: I had to adjust quite a few things with the xml format. Maybe something is different in the xmlschema version I use?

Comment (QuantumChemist) [reply adis_tools/schemas/qes_230310.xsd]: ofc now some tests fail... 🙄

File: adis_tools/schemas/qes_230310.xsd [hunk 3]
```diff
@@ -150,8 +150,8 @@ datecode 220603
       <element type="double" name="press_conv_thr" default="5e-1"/>
       <element type="qes:lowhighType" name="verbosity" default="low"/>
       <element type="positiveInteger" name="print_every"/>
-       <element type="boolean" name="fcp"/>
-      <element type="boolean" name="rism"/>
+      <!-- <element type="boolean" name="fcp" default="false"/>
```
Comment (jan-janssen) [inline adis_tools/schemas/qes_230310.xsd]: Removing this line causes the pyiron tests to fail with:
```
Reason: Unexpected child with tag 'fcp' at position 18.
```
To me there seems to be an issue with the input files.

File: adis_tools/schemas/qes_230310.xsd (hunk 2 above)
Comment (jan-janssen) [reply adis_tools/schemas/qes_230310.xsd]: I removed the modifications to the parser in https://github.com/materialdigital/ADIS2023/pull/19 and everything works fine. In addition, when I execute the notebook here locally I also get an `xml` related `Reason: Unexpected child with tag 'fcp' at position 18.` error, but the notebook execution does not fail. It seems like `jobflow` is converting the error to a warning and continues executing the remaining cells of the notebook. You can test it with [Binder](https://mybinder.org/v2/gh/QuantumChemist/ADIS2023/HEAD)

Comment (jan-janssen) [reply adis_tools/schemas/qes_230310.xsd]: Did you test it locally with a different quantum espresso version? In the repository we currently use `qe=7.2`.

Comment (QuantumChemist) [reply adis_tools/schemas/qes_230310.xsd]: I think it's the qe version as I used qe 6.7. Thank you for looking into this and fixing this issue 😄 
I will merge your commits back into mine to have passing tests in my repo.

**Review by jan-janssen — APPROVED** [review]
Looks good to me

**Comment by JaGeo** [comment]
@jan-janssen I will give it another read this week. Alternatively, we can merge this for now and I make a potential new pull request with improvements.

(ToDo for tomorrow)

**Comment by jan-janssen** [comment]
> @jan-janssen I will give it another read this week. Alternatively, we can merge this for now and I make a potential new pull request with improvements.
> 
> (ToDo for tomorrow)

Both is fine for me, I still was not able to reach @mbercx so far ...

**Comment by QuantumChemist** [comment]
> @jan-janssen I will give it another read this week. Alternatively, we can merge this for now and I make a potential new pull request with improvements.
> 
> (ToDo for tomorrow)

I'd say the second option is easier. Then we don't have to merge/fork back and forth 😆

**Comment by JaGeo** [comment]
Then, I will merge!

File: adis_tools/schemas/qes_230310.xsd (hunk 2 above)
Comment (QuantumChemist) [reply adis_tools/schemas/qes_230310.xsd]: I also realized that I overlooked the environment.yml file 🙈

**Comment by JaGeo** [comment]
@QuantumChemist just saw that there is still old text on the structure in here. We use MSONatoms instead of Structure. I will correct it

**Comment by QuantumChemist** [comment]
> @QuantumChemist just saw that there is still old text on the structure in here. We use MSONatoms instead of Structure. I will correct it

Oh that makes sense. I was a bit confused about it, but somehow didn't further think about it 🙈

**Pull request merged**